tensorflow_lite_support/python/task/processor/proto/segmentations_pb2.py

```diff
@@ -13,7 +13,207 @@
+    where higher values represent a stronger confidence.
+    Ultimately this is model specific, and other range of values might be
+    used.
+
+  Attributes:
+    value: The value indicates the prediction confidence usually in the
+      range [0, 1].
+  """
+  value: float
+
+  @doc_controls.do_not_generate_docs
+  def to_pb2(self) -> _ConfidenceMask:
+    """Generates a protobuf object to pass to the C++ layer."""
+    return _ConfidenceMask(value=self.value)
+
+  @classmethod
+  @doc_controls.do_not_generate_docs
+  def create_from_pb2(cls, pb2_obj: _ConfidenceMask) -> "ConfidenceMask":
+    """Creates a `ConfidenceMask` object from the given protobuf object."""
+    return ConfidenceMask(value=pb2_obj.value)
+
+  def __eq__(self, other: Any) -> bool:
+    """Checks if this object is equal to the given object.
+    Args:
+      other: The object to be compared with.
+    Returns:
+      True if the objects are equal.
+    """
+    if not isinstance(other, ConfidenceMask):
+      return False
+
+
+@dataclasses.dataclass
+class ColoredLabel:
+  """Defines a label associated with an RGB color, for display purposes.
+
+  Attributes:
+    color: The RGB color components for the label, in the [0, 255] range.
+    category_name: The class name, as provided in the label map packed in the
+      TFLite ModelMetadata.
+    display_name: The display name, as provided in the label map
+      (if available) packed in the TFLite Model Metadata .
+  """
+
+  color: Tuple[int, int, int]
+  category_name: str
+  display_name: str
+
+  @doc_controls.do_not_generate_docs
+  def to_pb2(self) -> _ColoredLabel:
+    """Generates a protobuf object to pass to the C++ layer."""
+    r, g, b = self.color
+    return _ColoredLabel(r=r, g=g, b=b,
+                         class_name=self.category_name,
+                         display_name=self.display_name)
+
+  @classmethod
+  @doc_controls.do_not_generate_docs
+  def create_from_pb2(cls, pb2_obj: _ColoredLabel) -> "ColoredLabel":
+    """Creates a `ColoredLabel` object from the given protobuf object."""
+    return ColoredLabel(color=(pb2_obj.r, pb2_obj.g, pb2_obj.b),
+                        category_name=pb2_obj.class_name,
+                        display_name=pb2_obj.display_name)
+
+  def __eq__(self, other: Any) -> bool:
+    """Checks if this object is equal to the given object.
+    Args:
+      other: The object to be compared with.
+    Returns:
+      True if the objects are equal.
+    """
+    if not isinstance(other, ColoredLabel):
+      return False
+
+
+@dataclasses.dataclass
+class Segmentation:
+  """Represents one Segmentation object in the image segmenter's results.
+
+  Attributes:
+    colored_labels: A list of `ColoredLabel` objects.
+    category_mask: A bytearray of the category mask.
+    confidence_masks: A list of `ConfidenceMask` objects.
+  """
+
+  colored_labels: List[ColoredLabel]
+  category_mask: bytearray = None
```

Comment:
Can you convert this to a 2D numpy array so that it's easier for the library user to handle?

Reply:
@khanhlvg Wouldnt it be time-consuming to reshape the array to a 2D one? It seems to take a long time to reshape for me. For now, I've kept it as a flattened NumPy array.

Reply:
Reshape is a very cheap operation. For example, reshaping a 1,000,000 flat array into a 1000 * 1000 2D array only takes 0.006 ms. Please return all masks as 2D array instead of flat array.